packages/camera/ios/Classes/CameraPlugin.m

```diff
@@ -148,17 +160,85 @@ - (void)captureToFile:(NSString *)path result:(FlutterResult)result {
 - (void)captureOutput:(AVCaptureOutput *)output
     didOutputSampleBuffer:(CMSampleBufferRef)sampleBuffer
            fromConnection:(AVCaptureConnection *)connection {
-  CVPixelBufferRef newBuffer = CMSampleBufferGetImageBuffer(sampleBuffer);
-  CFRetain(newBuffer);
-  CVPixelBufferRef old = _latestPixelBuffer;
-  while (!OSAtomicCompareAndSwapPtrBarrier(old, newBuffer, (void **)&_latestPixelBuffer)) {
-    old = _latestPixelBuffer;
+  if (output == _captureVideoOutput) {
+    CVPixelBufferRef newBuffer = CMSampleBufferGetImageBuffer(sampleBuffer);
+    CFRetain(newBuffer);
+    CVPixelBufferRef old = _latestPixelBuffer;
+    while (!OSAtomicCompareAndSwapPtrBarrier(old, newBuffer, (void **)&_latestPixelBuffer)) {
+      old = _latestPixelBuffer;
+    }
+    if (old != nil) {
+      CFRelease(old);
+    }
+    if (_onFrameAvailable) {
+      _onFrameAvailable();
+    }
+  }
+  if (!CMSampleBufferDataIsReady(sampleBuffer)) {
+    _eventSink(@{
+      @"event" : @"error",
+      @"errorDescription" : @"sample buffer is not ready. Skipping sample"
+    });
+    return;
+  }
+  if (_videoWriter.status == AVAssetWriterStatusFailed) {
+    _eventSink(@{
+      @"event" : @"error",
+      @"errorDescription" : @"AVAssetWriter failed! File may already exist at given path."
```

Comment:
I think you can give a better errormessage by looking in _videoWriter.error

Reply:
Done